tests/integration/docidcache/test_crud_cache.py

```diff
@@ -0,0 +1,314 @@
+import os
+from pathlib import Path
+
+import numpy as np
+import pytest
+
+from jina import Flow, Document
+from jina.executors.indexers import BaseIndexer
+from jina.executors.indexers.cache import DocIDCache
+from jina.executors.indexers.keyvalue import BinaryPbIndexer
+from jina.executors.indexers.vector import NumpyIndexer
+
+cur_dir = Path(os.path.dirname(os.path.abspath(__file__)))
+
+KV_IDX_FILENAME = 'kv_idx.bin'
+VEC_IDX_FILENAME = 'vec_idx.bin'
+DOCS_TO_SEARCH = 1
+TOP_K = 5
+BATCH_SIZE = 4
+DOCS_TO_INDEX = 10
+
+pytest.register_assert_rewrite('check_indexers_size')
+
+
+def config_env(field, tmp_workspace, shards, indexers, polling):
+    os.environ['JINA_SHARDS'] = str(shards)
+    os.environ['JINA_CACHE_FIELD'] = field
+    os.environ['JINA_POLLING'] = polling
+    os.environ['JINA_TOPK'] = str(TOP_K)
+    os.environ['JINA_TEST_CACHE_CRUD_WORKSPACE'] = str(tmp_workspace)
+    os.environ['JINA_KV_IDX_NAME'] = KV_IDX_FILENAME.split('.bin')[0]
+    os.environ['JINA_VEC_IDX_NAME'] = VEC_IDX_FILENAME.split('.bin')[0]
+    if indexers == 'parallel':
+        # the second indexer will be directly connected to entry gateway
+        os.environ['JINA_KV_NEEDS'] = 'cache'
+        os.environ['JINA_MERGER_NEEDS'] = '[vector, kv]'
+    else:
+        # else it requires to be in serial connection, after the first indexer
+        os.environ['JINA_KV_NEEDS'] = 'vector'
+        os.environ['JINA_MERGER_NEEDS'] = 'kv'
+
+
+np.random.seed(0)
+d_embedding = np.array([1, 1, 1, 1, 1, 1, 1])
+c_embedding = np.array([2, 2, 2, 2, 2, 2, 2])
+
+
+def get_documents(chunks, same_content, nr=10, index_start=0):
+    next_chunk_id = nr + index_start
+    for i in range(index_start, nr + index_start):
+        with Document() as d:
+            d.id = i
+            if same_content:
+                d.text = 'hello world'
+                d.embedding = d_embedding
+            else:
+                d.text = f'hello world {i}'
+                d.embedding = np.random.random(d_embedding.shape)
+            for j in range(chunks):
+                with Document() as c:
+                    c.id = next_chunk_id
+                    if same_content:
+                        c.text = 'hello world from chunk'
+                        c.embedding = c_embedding
+                    else:
+                        c.text = f'hello world from chunk {j}'
+                        c.embedding = np.random.random(d_embedding.shape)
+
+                next_chunk_id += 1
+                d.chunks.append(c)
+        yield d
+
+
+docs_chunks = [0, 3, 5]
```

Comment:
this is not used anymore ??

Reply:
Right, will remove. Thanks. They're declared in-line